src/components/map/Map.tsx

```diff
@@ -48,21 +51,43 @@ export const Map = (props: MapCoordinateState) => {
     const inputGeoCoder = new MapboxGeocoder({
       accessToken: mapboxgl.accessToken,
       mapboxgl: mapboxgl
-      })
+    })
 
     // Add geolocate control to the map.
-    map.addControl(locateMeGeocoder);
     locateMeGeocoder.on('geolocate', setUserLocation);
-    locateMeGeocoder.on('trackuserlocationstart', setUserLocation);
+    locateMeGeocoder.on('geolocate', queryRiskLevel)
     locateMeGeocoder.on('error', (e: Error) => console.error(e))
+    map.addControl(locateMeGeocoder);
 
     // Add input control to map
-    map.addControl(inputGeoCoder) 
+    inputGeoCoder.on('result', setUserLocation)
+    inputGeoCoder.on('result', (e: any) => queryRiskLevel({coords: {longitude: e.result.center[0], latitude: e.result.center[1]}}))
+    map.addControl(inputGeoCoder)
 
   }, [props.lat, props.lng])
 
-  function setUserLocation(e: any) {
+  async function setUserLocation(e: any) {
     (map.getSource('single-point') as any).setData(e.data);
+
+  }
+
+  async function queryRiskLevel(e: any) {
+
+    const { latitude, longitude } =  e && e.coords ? e.coords : { latitude: 0, longitude: 0 }
+    
+    const riskLevel = await riskLevelService.riskLevelByGeoPos(latitude, longitude)
```

Comment:
What happens if we request a risk level for the location we don't have data for? Does this throw an error? Should we process it?

Reply:
we will definitely get errors back, either from mapgl calls while the map is panning or from our own api. 

Throwing an error here will stop execution, until we add error handling by awaiting the end of a pan run, or add error messaging for the user.

For now just put proper error reporting so we an tell which of the errors on the console we expected to catch and which we didn't.